terraform/transform_provider.go

```diff
@@ -195,8 +195,8 @@ func (t *MissingProviderTransformer) Transform(g *Graph) error {
 				continue
 			}
 
-			// Add our own missing provider node to the graph
-			m[p] = g.Add(&graphNodeMissingProvider{ProviderNameValue: p})
+			// Add the missing provider node to the graph
+			m[p] = g.Add(&graphNodeProvider{ProviderNameValue: p})
```

Comment:
I'm actually +1 on this rename or something similar (since this node really does represent the provider itself in the graph) but I'd prefer we do it as a separate PR.